### DISCUSSION TITLE
Specific condition on the schema method called transform

### DISCUSSION THREAD

**Discussion by zevjr** [colinhacks/zod#3034] [Q&A] [answered]
When using the transform method, to exchange the keys of an object after it is validated.
```typescript
const SalesOrderProductSchema = z.object({
  integration_order_code: z.string().max(60),
  order_code: z.number().optional(),
  order_number: z.string().max(15).optional(),
}).transform((data) => ({
  code_pedido_integracao: data.integration_order_code,
  codigo_pedido: data.order_code,
  numero_pedido: data.order_number,
}))
```

At the output of this action, it was expected that the object would have the name of the keys equal to the items that are within the transform method.
however, this does not happen, the object is returning the same as the input object and there is no error occurring.

Is the form presented correct?
Is there any other way to get the expected result? or is it a bug found?

**Comment by JacobWeisenburger** [comment, marked as answer]
I have tested this code and everything is working as expected. What is the problem you are having?
```ts
const SalesOrderProductSchema = z.object( {
    integration_order_code: z.string().max( 60 ),
    order_code: z.number().optional(),
    order_number: z.string().max( 15 ).optional(),
} ).transform( ( data ) => ( {
    code_pedido_integracao: data.integration_order_code,
    codigo_pedido: data.order_code,
    numero_pedido: data.order_number,
} ) )

type Data = z.infer<typeof SalesOrderProductSchema>
// type Data = {
//     code_pedido_integracao: string
//     codigo_pedido: number | undefined
//     numero_pedido: string | undefined
// }

console.log( SalesOrderProductSchema.parse( {
    integration_order_code: '123',
    order_code: 123,
    order_number: '123',
} ) )
// {
//     code_pedido_integracao: "123",
//     codigo_pedido: 123,
//     numero_pedido: "123"
// }
```

If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger
